### DISCUSSION TITLE
Hy and MongoEngine

### DISCUSSION THREAD

**Discussion by RickBradford** [hylang/hy#2316] [Q&A] [answered]
Is it possible to create a MongoDB/MongoEngine schema in Hy?

The MongoEngine manual gives a typical Python schema as 

```
class User(Document):
    email = StringField(required=True)
    first_name = StringField(max_length=50)
    last_name = StringField(max_length=50)
```
which can be instantiated as 

`ross = User(email='ross@example.com', first_name='Ross', last_name='Lawley')`

What would the equivalent be in Hy (if it is possible to do)?

**Comment by Kodiologist** [comment, marked as answer]
It should always be possible to do a straight translation of any Python code into Hy. In this case, it's:

```
(defclass User [Document]
  (setv email (StringField :required True))
  (setv first_name (StringField :max_length 50))
  (setv last_name (StringField :max_length 50)))

(setv ross (User :email "ross@example.com" :first_name "Ross" :last_name "Lawley"))
```

**Reply by RickBradford** [reply]
OK, thanks for the clarification.
